CorsixTH/Lua/entities/humanoids/patient.lua

```diff
@@ -301,8 +301,13 @@ function Patient:isTreatmentEffective()
   local cure_chance = self.hospital.disease_casebook[self.disease.id].cure_effectiveness
   cure_chance = cure_chance * self.diagnosis_progress
 
-  local die = self.die_anims and math.random(1, 100) > cure_chance
-  return not die
+  -- Service quality has a factor on cure chance +/- 10%
+  local room = self:getRoom()
+  local base, divisor = 0.9, 5
+  local service_factor = base + (room:getStaffServiceQuality() / divisor)
+  cure_chance = cure_chance * service_factor
```

Comment:
So when I have a bad treatment the service quality has little impact, while if I have a good treatment the service quality has a lot of impact.
So effectively it's shortening the last part of the trajectory to reach an ultimate treatment with no deaths occurring.

Have you considered affecting the chance  by adding some of the remaining chance instead? That is, use `1 - cure_chance` as multiplication factor for the service_factor (and adding that to the chance)?
Then service quality has a larger effect to a bad treatment, and not much impact on a good treatment.

Reply:
You'd need 100 - cure_chance. But also cure chance can be >100

Reply:
`math.max(100 - cure_chance, 0)` thus